models/classes/runner/RunnerService.php

```diff
@@ -184,10 +184,11 @@ public function getItemSession(RunnerServiceContext $context);
      * @param $direction
      * @param $scope
      * @param $ref
+     * @param string $toolStates
      * @return boolean
      * @throws \common_Exception
      */
-    public function move(RunnerServiceContext $context, $direction, $scope, $ref);
+    public function move(RunnerServiceContext $context, $direction, $scope, $ref, $toolStates);
```

Comment:
I don't think the tool state should be part of the `move` there will be other places we will synchronize the state : `skip` to be verfiied, `timeout` for sure, `exitTest`, etc. everywhere we save the response and/or the item state.
May be add 2 new methods `setToolState` and `getToolState`